From 77deb26ced5396522e6dedfb2ff057ab0a7a3aba Mon Sep 17 00:00:00 2001
Message-Id: <77deb26ced5396522e6dedfb2ff057ab0a7a3aba.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Thu, 31 Jul 2014 17:59:50 +0530
Subject: [PATCH 166/312] FOR_UPSTREAM [VPG]: drm/i915: Change
 mode_config.mutex to mode_config.connection_mutex
 in DPST routine as per forklift design.

intel_get_pipe_from_connector expects mode_config.connection_mutex
to be locked by the caller. Patch changes mode_config.mutex to
mode_config.connection_mutex in DPST routine as per forklift design

Issue: GMIN-2839
Change-Id: I0c79dcff5b8fce0c2319ea5af4e737bf674a5f59
Signed-off-by: Akash Goel <akash.goel@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/i915_dpst.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_dpst.c b/drivers/gpu/drm/i915/i915_dpst.c
index e294ed4c..1698845 100644
--- a/drivers/gpu/drm/i915/i915_dpst.c
+++ b/drivers/gpu/drm/i915/i915_dpst.c
@@ -230,11 +230,11 @@ i915_dpst_apply_luma(struct drm_device *dev,
 	ioctl_data->ie_container.dpst_blc_factor;
 
 	/* Avoid warning messages */
-	mutex_lock(&dev->mode_config.mutex);
+	drm_modeset_lock(&dev->mode_config.connection_mutex, NULL);
 	mutex_lock(&dev_priv->backlight_lock);
 	i915_dpst_set_brightness(dev, panel->backlight.level);
 	mutex_unlock(&dev_priv->backlight_lock);
-	mutex_unlock(&dev->mode_config.mutex);
+	drm_modeset_unlock(&dev->mode_config.connection_mutex);
 
 	/* Enable Image Enhancement Table */
 	blm_hist_ctl = I915_READ(dev_priv->dpst.reg.blm_hist_ctl);
-- 
1.7.9.5

